pkg/controller/windowsmachineconfig/windows/windows.go

```diff
@@ -51,6 +51,17 @@ func (vm *Windows) Configure() error {
 	return vm.runBootstrapper()
 }
 
+// validate the WindowsVM node object
+func (vm *Windows) Validate() error {
+	if vm.GetCredentials() == nil {
```

Comment:
You are using `vm.GetCredentials().GetInstanceId()` in multiple places without validating it here. Please add that.